packages/@aws-cdk/aws-eks/lib/helm-chart.ts

```diff
@@ -0,0 +1,122 @@
+import { CustomResource, CustomResourceProvider } from '@aws-cdk/aws-cloudformation';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct, Duration, Stack } from '@aws-cdk/core';
+import * as path from 'path';
+import { Cluster } from './cluster';
+import { KubectlLayer } from './kubectl-layer';
+
+/**
+ * Helm Chart options.
+ */
+
+export interface HelmChartOptions {
+  /**
+   * The name of the release.
+   */
+  readonly release?: string;
+
+  /**
+   * The name of the chart.
+   */
+  readonly chart: string;
+
+  /**
+   * The chart version to install.
+   * @default - If this is not specified, the latest version is installed
+   */
+  readonly version?: string;
+
+  /**
+   * The repository which contains the chart. For example: https://kubernetes-charts.storage.googleapis.com/
+   * @default - No repository will be used, which means that the chart needs to be an absolute URL.
+   */
+  readonly repository?: string;
+
+  /**
+   * The Kubernetes namespace scope of the requests.
+   * @default default
+   */
+  readonly namespace?: string;
+
+  /**
+   * The values to be used by the chart.
+   * @default - No values are provided to the chart.
+   */
+  readonly values?: {[key: string]: any};
+}
+
+/**
+ * Helm Chart properties.
+ */
+export interface HelmChartProps extends HelmChartOptions {
+  /**
+   * The EKS cluster to apply this configuration to.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+}
+
+/**
+ * Represents a helm chart within the Kubernetes system.
+ *
+ * Applies/deletes the resources using `kubectl` in sync with the resource.
+ */
+export class HelmChart extends Construct {
+  /**
+   * The CloudFormation reosurce type.
+   */
+  public static readonly RESOURCE_TYPE = 'Custom::AWSCDK-EKS-HelmChart';
+
+  constructor(scope: Construct, id: string, props: HelmChartProps) {
+    super(scope, id);
+
+    const stack = Stack.of(this);
+
+    // we maintain a single manifest custom resource handler for each cluster
+    const handler = this.getOrCreateHelmChartHandler(props.cluster);
+    if (!handler) {
+      throw new Error(`Cannot define a Helm chart on a cluster with kubectl disabled`);
+    }
+
+    new CustomResource(this, 'Resource', {
+      provider: CustomResourceProvider.lambda(handler),
+      resourceType: HelmChart.RESOURCE_TYPE,
+      properties: {
+        Release: props.release || this.node.uniqueId.toLowerCase(),
```

Comment:
If this value is limited in length, you should trim-left it (the tail end includes the hash)

Reply:
Done and added a test to confirm.